packages/report-server/src/routes/FetchReportRoute.ts

```diff
@@ -32,55 +32,34 @@ export class FetchReportRoute extends Route<FetchReportRequest> {
     return report;
   }
 
-  async checkUserHasAccessToReport(
-    report: ReportType,
-    hierarchy = 'explore',
-    requestedOrgUnitCodes: string[],
-  ) {
-    const { accessPolicy, ctx } = this.req;
-    const permissionGroupName = await report.permissionGroupName();
-
-    const foundOrgUnits = await ctx.microServices.entityApi.getEntities(
-      hierarchy,
-      requestedOrgUnitCodes,
-      {
-        fields: ['code', 'country_code'],
-      },
-    );
-    const foundOrgUnitCodes = foundOrgUnits.map(orgUnit => orgUnit.code);
-
-    const missingOrgUnitCodes = requestedOrgUnitCodes.filter(
-      orgUnitCode => !foundOrgUnitCodes.includes(orgUnitCode),
-    );
-    if (missingOrgUnitCodes.length > 0) {
-      throw new Error(`No entities found with codes ${missingOrgUnitCodes}`);
-    }
-
-    const countryCodes = new Set(foundOrgUnits.map(orgUnit => orgUnit.country_code));
-    countryCodes.forEach(countryCode => {
-      if (countryCode === null || !accessPolicy.allows(countryCode, permissionGroupName)) {
-        throw new Error(`No ${permissionGroupName} access for user to ${countryCode}`);
-      }
-    });
-  }
-
   async buildResponse() {
     const { query, body } = this.req;
-    const { organisationUnitCodes, hierarchy, ...restOfParams } = { ...query, ...body };
+    const { organisationUnitCodes, hierarchy = 'explore', ...restOfParams } = { ...query, ...body };
     if (!organisationUnitCodes) {
       throw new Error('Must provide organisationUnitCodes URL parameter');
     }
     const report = await this.findReport();
+    const permissionGroupName = await report.permissionGroupName();
 
-    const organisationUnitCodesArray = Array.isArray(organisationUnitCodes)
+    const requestedOrgUnitCodesArray = Array.isArray(organisationUnitCodes)
```

Comment:
This we can move this logic to define 'requestedOrgUnitCodesArray' into the `getRequestedOrgUnitObjects` function? That way we don't need to do it here and in `TestReportRoute`?